src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -127,33 +124,45 @@ public IList<VSTestResult> GetVSTestResults(XmlNode resultNode)
         /// </summary>
         private TestCase MakeTestCaseFromXmlNode(XmlNode testNode)
         {
+            var methodName = testNode.GetAttribute("methodname");
+            var displayName = testNode.GetAttribute("name");
+            var className = testNode.GetAttribute("classname");
+
             var testCase = new TestCase(
                                      testNode.GetAttribute("fullname"),
                                      new Uri(NUnitTestAdapter.ExecutorUri),
                                      _sourceAssembly)
             {
-                DisplayName = testNode.GetAttribute("name"),
+                DisplayName = displayName,
                 CodeFilePath = null,
                 LineNumber = 0
             };
 
             if (_collectSourceInformation && _navigationDataProvider != null)
             {
-                var className = testNode.GetAttribute("classname");
-                var methodName = testNode.GetAttribute("methodname");
                 var navData = _navigationDataProvider.GetNavigationData(className, methodName);
                 if (navData.IsValid)
                 {
                     testCase.CodeFilePath = navData.FilePath;
                     testCase.LineNumber = navData.LineNumber;
+                }               
+            }
+            else
+            {
+                var testName = testNode.GetAttribute("TestName");
+
+                if (!string.IsNullOrEmpty(testName))
```

Comment:
I don't understand this. Where do you expect the `TestName` attribute to come from?

Reply:
Added 'testname' attribute in test case xml. Please check PR: https://github.com/nunit/nunit/pull/2657/files